watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1175 @@
+                    "maxRelViscosityOfDispersionAtInversion": relative_visc_inversion,
+                }
+            )
+        return config
+
+    def _add_to_json(
         self,
         flash_method,
-        state_vars,
-        water_analysis_output=None,
-        inflows_phase="total",
-        use_scaling_rigorous=True,
+        input_data,
+        included_solids,
+        excluded_solids,
+        use_scaling_rigorous,
+        file_name,
     ):
         """
-        Builds a base dictionary required for OLI flash analysis.
-
-        :param flash_method: string name of OLI flash flash_method to use
-        :param state_vars: dictionary containing solutes, temperatures, pressure, and units
-        :param water_analysis_output: dictionary to extract inflows from (required if not wateranalysis flash)
-        :param inflows_phase: string indicating desired phase of inflows
-        :param use_scaling_rigorous: boolean switch to use estimated or rigorous solving for prescaling metrics
-
-        :return inputs: dictionary containing inputs for specified OLI flash analysis
+        Add input data to JSON.
+
+        :param flash_method: string for flash calculation name
+        :param input_data: data object from flash configuration function
+        :param included_solids: list of solids to include in analysis
+        :param excluded_solids: list of solids to exclude from analysis
+        :param use_scaling_rigorous: bool to switch between Rigorous (default) and Estimated scaling computations
+        :param file_name: string for file to write, if any
         """
 
-        inputs = {"params": {}}
+        self._set_prescaling_calculation_mode(use_scaling_rigorous)
+        json_input = {"params": {}}
 
         if flash_method == "wateranalysis":
-            inputs["params"] = {
-                "waterAnalysisInputs": self._build_input_list(state_vars)
-            }
+            json_input["params"].update({"waterAnalysisInputs": input_data})
         else:
-            if not water_analysis_output:
-                raise IOError(
-                    "Run wateranalysis flash to generate water_analysis_output data."
-                )
+            json_input["params"] = input_data
 
-            inputs["params"] = {
-                "temperature": {
-                    "unit": str(state_vars["units"]["temperature"]),
-                    "value": float(state_vars["temperature"]),
-                },
-                "pressure": {
-                    "unit": str(state_vars["units"]["pressure"]),
-                    "value": float(state_vars["pressure"]),
-                },
-                "inflows": self._extract_inflows(water_analysis_output, inflows_phase),
-            }
+        additional_params = {
+            "optionalProperties": dict(self.optional_properties),
+            "unitSetInfo": dict(self.output_unit_set),
+        }
+        if included_solids and excluded_solids:
```

Comment:
Is there a reason why both cannot be specified at the same time?

Reply:
As per OLI: 

> params.excludedSolids and params.includedSolids cannot be specified at the same time.